src/coreclr/jit/hwintrinsic.h

```diff
@@ -926,6 +930,39 @@ struct HWIntrinsicInfo
         return (flags & HW_Flag_HasEnumOperand) != 0;
     }
 
+    static bool HasScalarVariant(NamedIntrinsic id)
+    {
+        const HWIntrinsicFlag flags = lookupFlags(id);
+        return (flags & HW_Flag_HasScalarVariant) != 0;
+    }
+
+    static NamedIntrinsic GetScalarVariant(NamedIntrinsic id)
+    {
+        switch (id)
```

Comment:
might be good to add `assert(HasScalarVariant(id));` so we can see the assert message instead of `unreached()`.